signer/contentsignaturepki/upload.go

```diff
@@ -26,21 +34,28 @@ func (s *ContentSigner) upload(data, name string) error {
 	}
 	switch parsedURL.Scheme {
 	case "s3":
-		return uploadToS3(data, name, parsedURL)
+		// Context is a required argument, but in our uses,
+		// LoadDefaultConfig pulls the necessary configuration
+		// from the environment.
+		cfg, err := config.LoadDefaultConfig(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to load AWS config: %w", err)
+		}
+		client := s3.NewFromConfig(cfg)
+		uploader := manager.NewUploader(client)
+		return uploadToS3(uploader, data, name, parsedURL)
 	case "file":
 		return writeLocalFile(data, name, parsedURL)
 	default:
 		return fmt.Errorf("unsupported upload scheme " + parsedURL.Scheme)
 	}
 }
 
-func uploadToS3(data, name string, target *url.URL) error {
-	sess := session.Must(session.NewSession())
-	uploader := s3manager.NewUploader(sess)
-	_, err := uploader.Upload(&s3manager.UploadInput{
+func uploadToS3(client S3UploadAPI, data, name string, target *url.URL) error {
+	_, err := client.Upload(context.Background(), &s3.PutObjectInput{
 		Bucket:             aws.String(target.Host),
 		Key:                aws.String(target.Path + name),
-		ACL:                aws.String("public-read"),
+		ACL:                "public-read",
```

Comment:
There's a constant we can use for this now: `types.ObjectCannedACLPublicRead` from `"github.com/aws/aws-sdk-go-v2/service/s3/types"`